msprime/simulations.py

```diff
@@ -1063,18 +1082,30 @@ def __str__(self):
 
 
 class SimulationModelChange(DemographicEvent):
-    # TODO document
+    """
+    An event representing a change of underlying simulation model.
+
+    :param float time: The time at which the simulation model changes
+        to the new model, in generations.
+    :param model: The new simulation model to use.
+        This can either be a string (e.g., ``"smc_prime"``) or an instance of
+        a simulation model class (e.g, ``msprime.DiscreteTimeWrightFisher(100)``.
+        Please see the :ref:`sec_api_simulation_models` section for more details
+        on specifying simulations models. If the argument is a string, the
+        reference population size is set from the top level ``Ne`` parameter
+        to :func:`.simulate`. If this is None (the default) we revert to
+        the standard coalescent with the reference population size set by ``Ne``.
```

Comment:
Could mention that default reference pop size is also set by Ne if model is specified but with default reference population size (if that's actually the case)?